The warning still exists:
```

  Warning: C:/actions-runner/opencv/opencv/opencv/modules/core/src/minmax.simd.hpp(276,1): warning C4293: '<<': shift count negative or too big, undefined behavior (compiling source file C:\actions-runner\opencv\opencv\build\modules\core\minmax.avx512_skx.cpp) [C:\actions-runner\opencv\opencv\build\modules\core\opencv_core_AVX512_SKX.vcxproj]
  Warning: C:/actions-runner/opencv/opencv/opencv/modules/core/src/minmax.simd.hpp(276,1): warning C4293: '<<': shift count negative or too big, undefined behavior (compiling source file C:\actions-runner\opencv\opencv\build\modules\core\minmax.avx2.cpp) [C:\actions-runner\opencv\opencv\build\modules\core\opencv_core_AVX2.vcxproj]
  Warning: C:/actions-runner/opencv/opencv/opencv/modules/core/src/minmax.simd.hpp(276,1): warning C4293: '<<': shift count negative or too big, undefined behavior (compiling source file C:\actions-runner\opencv\opencv\build\modules\core\minmax.sse4_1.cpp) [C:\actions-runner\opencv\opencv\build\modules\core\opencv_core_SSE4_1.vcxproj]
  Warning: C:/actions-runner/opencv/opencv/opencv/modules/core/src/minmax.simd.hpp(276,1): warning C4293: '<<': shift count negative or too big, undefined behavior (compiling source file C:\actions-runner\opencv\opencv\opencv\modules\core\src\minmax.dispatch.cpp) [C:\actions-runner\opencv\opencv\build\modules\core\opencv_core.vcxproj]
```

--- a/modules/core/src/minmax.simd.hpp
+++ b/modules/core/src/minmax.simd.hpp
@@ -248,6 +248,10 @@ static inline void mm_fold_max(const VT& valMax, const IT& idxMax, const IT& non
     }
 }
 
+template<typename IST> struct mm_idxcap { static const int64_t value = (int64_t)INT_MAX; };
+template<> struct mm_idxcap<uchar>  { static const int64_t value = (int64_t)UCHAR_MAX; };
+template<> struct mm_idxcap<ushort> { static const int64_t value = (int64_t)USHRT_MAX; };
+
 // IST = unsigned index lane type (uchar / ushort / uint / uint64_t).
 template<typename T, typename VT, typename IT, typename IST, typename WT>
 static void minMaxIdx_simd_(const T* src, const uchar* mask, WT* minval, WT* maxval,
@@ -273,7 +277,7 @@ static void minMaxIdx_simd_(const T* src, const uchar* mask, WT* minval, WT* max
             const IT none = mm_set<IT>((IST)~(IST)0);
             // Reduce before the per-lane index could reach the 'none' sentinel.
             // For >= 32-bit indices (len <= INT_MAX) one block covers everything.
-            const int64_t idxcap = (sizeof(IST) <= 2) ? (int64_t)std::numeric_limits<IST>::max() : (int64_t)INT_MAX;
+            const int64_t idxcap = mm_idxcap<IST>::value;
             const int blockStep = (int)((idxcap / nlanes) * nlanes);
 
             const IT inc2 = mm_set<IT>((IST)(nlanes * 2));

--- a/modules/core/src/minmax.simd.hpp
+++ b/modules/core/src/minmax.simd.hpp
@@ -248,10 +248,6 @@ static inline void mm_fold_max(const VT& valMax, const IT& idxMax, const IT& non
     }
 }
 
-template<typename IST> struct mm_idxcap { static const int64_t value = (int64_t)INT_MAX; };
-template<> struct mm_idxcap<uchar>  { static const int64_t value = (int64_t)UCHAR_MAX; };
-template<> struct mm_idxcap<ushort> { static const int64_t value = (int64_t)USHRT_MAX; };
-
 // IST = unsigned index lane type (uchar / ushort / uint / uint64_t).
 template<typename T, typename VT, typename IT, typename IST, typename WT>
 static void minMaxIdx_simd_(const T* src, const uchar* mask, WT* minval, WT* maxval,
@@ -277,7 +273,7 @@ static void minMaxIdx_simd_(const T* src, const uchar* mask, WT* minval, WT* max
             const IT none = mm_set<IT>((IST)~(IST)0);
             // Reduce before the per-lane index could reach the 'none' sentinel.
             // For >= 32-bit indices (len <= INT_MAX) one block covers everything.
-            const int64_t idxcap = mm_idxcap<IST>::value;
+            const int64_t idxcap = (sizeof(IST) <= 2) ? (int64_t)std::numeric_limits<IST>::max() : (int64_t)INT_MAX;
             const int blockStep = (int)((idxcap / nlanes) * nlanes);
 
             const IT inc2 = mm_set<IT>((IST)(nlanes * 2));

